dfx/index.ts

```diff
@@ -60,8 +60,38 @@ export function getIdentityName(): string {
     return execSync(`dfx identity whoami`).toString().trim();
 }
 
-export function generateIdentity(name: string) {
-    execSync(`dfx identity new ${name} --storage-mode plaintext`);
+type StorageMode = 'keyring' | 'password-protected' | 'plaintext';
+
+export function generateIdentity(
+    name: string,
+    storageModeCool?: StorageMode
+): Buffer {
+    console.info();
+    console.info(`Generating identity ${name}`);
+    const storageMode = determineStorageMode(storageModeCool);
+    if (storageMode === undefined) {
+        console.info(`You may be prompted to create a password for ${name}`);
+        console.info();
+        return execSync(`dfx identity new ${name}`, {
+            stdio: ['inherit', 'pipe', 'inherit']
+        });
+    }
+    if (storageMode === 'password-protected') {
+        console.info(`You will be prompted to create a password for ${name}`);
+        console.info();
+    }
+    return execSync(`dfx identity new ${name} --storage-mode ${storageMode}`, {
+        stdio: ['inherit', 'pipe', 'inherit'] // TODO add todo here and aboe
+    });
+}
+
+function determineStorageMode(
```

Comment:
Should we let the storage mode env var actually ser the storage mode?

And if it isn't one of the hard coded modes we can throw an error

Reply:
sure